### PR TITLE
Generate Route

### LINKED ISSUE #15: Display shortest route
Display the shortest route from A to B using a [Route API](https://github.com/pnancke/trip-planner/wiki/Routing-data-source#routing).

### PR DESCRIPTION
**Pull request by pnancke**
closes #15 
This Pull Requests contains some GUI changes, so remember to run the funtional tests manually.

### CODE REVIEW

File: src/integration-test/groovy/trip/planner/HomeFunctionalSpec.groovy [hunk 1]
```diff
@@ -37,4 +39,44 @@ class HomeFunctionalSpec extends GebSpec {
         then: "The autocorrect works"
         assert page.getAutoCompleteText(page.destinationDropdown).contains(LEIPZIG_FULLNAME)
     }
+
+    void "test generate route spinner exists"() {
```
Comment (MZober1993) [inline src/integration-test/groovy/trip/planner/HomeFunctionalSpec.groovy]: The given test works, but I got this error-message: "No bean named 'org.springframework.context.annotation.ConfigurationClassPostProcessor.importRegistry' is defined. Stacktrace follows:
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.grails.internal.SESSION_FACTORY_HOLDER': Initialization of bean failed; ..."

File: src/integration-test/groovy/trip/planner/HomeFunctionalSpec.groovy [hunk 2]
```diff
@@ -37,4 +39,44 @@ class HomeFunctionalSpec extends GebSpec {
...
+
+    void "test generate route spinner stops spinning"() {
+        when: "Valid cities are given"
+        HomePage page = browser.to HomePage
+        page.enterCity(page.startSearchField, LEIPZIG)
+        page.clickFirstAutocomplete(page.startDropdown)
+        page.enterCity(page.destinationSearchField, BERLIN)
+        page.clickFirstAutocomplete(page.destinationDropdown)
+        page.submitButton.click()
+        Thread.sleep(5000)
```
Comment (MZober1993) [inline src/integration-test/groovy/trip/planner/HomeFunctionalSpec.groovy]: please try to use waitFor for a faster test like this: 

`page.submitButton.click()

then: "spinner stops spinning"
assert waitFor{!page.spinnerExists()}`

File: src/integration-test/groovy/trip/planner/HomePage.groovy [hunk 3]
```diff
@@ -30,4 +33,14 @@ class HomePage extends Page {
...
+
+    void clickFirstAutocomplete(Navigator autoComplNav){
+        waitFor(10) { autoComplNav.displayed }
+        autoComplNav.click()
+    }
+
+    boolean spinnerExists() {
+        spinner
+    }
+
```
Comment (MZober1993) [inline src/integration-test/groovy/trip/planner/HomePage.groovy]: remove this line

File: grails-app/controllers/trip/planner/HomeController.groovy [hunk 4]
```diff
@@ -1,6 +1,39 @@
...
 
+import grails.plugins.rest.client.RestBuilder
+import groovy.json.JsonBuilder
+
 class HomeController {
 
-    def index() { }
+    def index() {}
+
+    def getRoute(String startLon, String startLat, String endLon, String endLat) {
```
Comment (MZober1993) [inline grails-app/controllers/trip/planner/HomeController.groovy]: Isn't startLon,... a double?

File: grails-app/controllers/trip/planner/HomeController.groovy [hunk 5]
```diff
@@ -1,6 +1,39 @@
...
+        def rest = new RestBuilder()
```
Comment (MZober1993) [inline grails-app/controllers/trip/planner/HomeController.groovy]: Please use explicit typs for complex objects (for a better safety) - and if you want you can use dynamic types for primitives.

File: grails-app/controllers/trip/planner/HomeController.groovy [hunk 6]
```diff
@@ -1,6 +1,39 @@
...
+        def url = "http://www.yournavigation.org/api/1.0/gosmore.php?format=kml&flat=$startLat&flon=$startLon&tlat=$endLat&&tlon=$endLon"
+        int tries = 1
+        def json = new JsonBuilder()
+        while (tries < 4) {
```
Comment (MZober1993) [inline grails-app/controllers/trip/planner/HomeController.groovy]: you can write this as a for loop and it works also (without using tries)

File: grails-app/controllers/trip/planner/HomeController.groovy [hunk 7]
```diff
@@ -1,6 +1,39 @@
...
+                def coordinates = xml.Document.Folder.Placemark.LineString.coordinates.text().trim().tokenize("\n")
+                def size = coordinates.size()
+                json {
+                    success true
+                    count size
+                    data coordinates.collect { it.tokenize(",") }
+                }
+                render json.toString()
+                return
+            } catch (Exception e) {
```
Comment (MZober1993) [inline grails-app/controllers/trip/planner/HomeController.groovy]: Gotta catch 'em all? Please add some testcases for this exception and then try to specify the real exception type.

File: src/integration-test/groovy/trip/planner/HomeFunctionalSpec.groovy [hunk 8]
```diff
@@ -37,4 +39,44 @@ class HomeFunctionalSpec extends GebSpec {
...
+    void "test generate route no route exists"() {
+        when: "Unconnectable cities are given"
+        HomePage page = browser.to HomePage
+        page.enterCity(page.startSearchField, LEIPZIG)
+        page.clickFirstAutocomplete(page.startDropdown)
+        page.enterCity(page.destinationSearchField, NEW_YORK)
+        page.clickFirstAutocomplete(page.destinationDropdown)
+
+        then: "Alert occurs"
+        withAlert {
```
Comment (MZober1993) [inline src/integration-test/groovy/trip/planner/HomeFunctionalSpec.groovy]: is it possible to use waitFor here?

File: grails-app/controllers/trip/planner/HomeController.groovy (hunk 6 above)
Comment (pnancke) [inline grails-app/controllers/trip/planner/HomeController.groovy]: Is there any advantage using a foor loop which does basically the same?

File: src/integration-test/groovy/trip/planner/HomeFunctionalSpec.groovy (hunk 1 above)
Comment (pnancke) [inline src/integration-test/groovy/trip/planner/HomeFunctionalSpec.groovy]: Could you please post the full stacktrace if you are able to reproduce this error?

**Pull request merged**